services/web/server/src/simcore_service_webserver/exporter/formatters/base_models.py

```diff
@@ -0,0 +1,111 @@
+from pathlib import Path
+from typing import Dict, Any, Union
+
+import aiofiles
+from pydantic import BaseModel, Field, validator
+
+
+class BaseModelSavePath(BaseModel):
+    root_dir: Path = Field(
+        ...,
+        description="temporary directory where all data is stored, to be ignored from serialization",
+    )
+    path_in_root_dir: Path = Field(
+        ...,
+        description="path to the file where to store this model, to be ignored from serialization",
+    )
+
+    @validator("root_dir")
+    @classmethod
+    def _validate_root_dir(cls, v):
+        if not isinstance(v, Path):
+            v = Path(v)
+        if not v.is_dir():
+            raise ValueError(f"Provided path {str(v)} is not a directory!")
+        return v
+
+    @validator("path_in_root_dir")
+    @classmethod
+    def _validate_path_in_root_dir(cls, v):
+        if not isinstance(v, Path):
+            v = Path(v)
+        if v.is_absolute():
+            raise ValueError(f"Must provide a relative path, not {str(v)}")
+
+        return v
+
+    @property
+    def path(self) -> Path:
+        return self.root_dir / self.path_in_root_dir
+
+    def is_store_path_present(self) -> bool:
+        """Used to check if the file is present in the expected location"""
+        return self.path.exists()
+
+    async def data_from_file(self) -> str:
+        async with aiofiles.open(self.path, "r") as input_file:
+            return await input_file.read()
+
+    async def data_to_file(self, payload: str) -> None:
+        async with aiofiles.open(self.path, "w") as output_file:
+            await output_file.write(payload)
+
+
+class BaseLoadingModel(BaseModel):
+    """
+    Used as base for all models which need to be validated to a file.
+    Serialization is done by Pydantic in json format.
+    """
+
+    # path relative to the export folder
+    _RELATIVE_STORAGE_PATH: Union[str, Path] = ""
```

Comment:
No, I need both, as I like to have the possibility to easily express the relative path with a string "some/path/to/file", this will be appended to the root path where all is stored, and as Path can both support Path and str for **"/" operation**, this works fine.